From bffc008323ac105a5e134918c80a0d8e21319dc7 Mon Sep 17 00:00:00 2001
From: Sun ChuanHu <aaron.sun@rock-chips.com>
Date: Fri, 10 Aug 2018 10:55:58 +0800
Subject: [PATCH] ARM: dts: rockchip:
 rk3308-voice-module-amic-mainboard-v10-aarch32 support blueetooth alsa card

Change-Id: I7910c76366edf40bc808f71d35a66402ce328ff7
Signed-off-by: Sun ChuanHu <aaron.sun@rock-chips.com>
---
 ...08-voice-module-amic-mainboard-v10-aarch32.dts |  4 ++++
 ...rk3308-voice-module-mainboard-v10-aarch32.dtsi |  2 +-
 .../boot/dts/rk3308-voice-module-v10-aarch32.dtsi | 15 +++++++++++++++
 3 files changed, 20 insertions(+), 1 deletion(-)

diff --git a/arch/arm/boot/dts/rk3308-voice-module-amic-mainboard-v10-aarch32.dts b/arch/arm/boot/dts/rk3308-voice-module-amic-mainboard-v10-aarch32.dts
index 5dcc61371bc4..770d4e91bd2b 100644
--- a/arch/arm/boot/dts/rk3308-voice-module-amic-mainboard-v10-aarch32.dts
+++ b/arch/arm/boot/dts/rk3308-voice-module-amic-mainboard-v10-aarch32.dts
@@ -18,6 +18,10 @@
 	rockchip,en-always-grps = <1 2>;
 };
 
+&bluetooth_sound {
+	status = "okay";
+};
+
 &is31fl3236 {
 	reg = <0x3f>;
 };
diff --git a/arch/arm/boot/dts/rk3308-voice-module-mainboard-v10-aarch32.dtsi b/arch/arm/boot/dts/rk3308-voice-module-mainboard-v10-aarch32.dtsi
index 8d70f702402e..36b1defbb57c 100644
--- a/arch/arm/boot/dts/rk3308-voice-module-mainboard-v10-aarch32.dtsi
+++ b/arch/arm/boot/dts/rk3308-voice-module-mainboard-v10-aarch32.dtsi
@@ -41,7 +41,7 @@
 	status = "okay";
 	rockchip,no-deep-low-power;
 	/* ADC3-6 for 4ch capture, loopback to ADC7 */
-	rockchip,adc-grps-route = <1 2 3 0>;
+	rockchip,adc-grps-route = <2 3 1 0>;
 };
 
 &i2c1 {
diff --git a/arch/arm/boot/dts/rk3308-voice-module-v10-aarch32.dtsi b/arch/arm/boot/dts/rk3308-voice-module-v10-aarch32.dtsi
index 92a27328cd32..7fe39a7ee84a 100644
--- a/arch/arm/boot/dts/rk3308-voice-module-v10-aarch32.dtsi
+++ b/arch/arm/boot/dts/rk3308-voice-module-v10-aarch32.dtsi
@@ -97,6 +97,17 @@
 		status = "disabled";
 	};
 
+	bluetooth_sound: bluetooth-sound {
+		status = "disabled";
+		compatible = "rockchip,multicodecs-card";
+		rockchip,card-name = "rockchip,rk3308-pcm";
+		rockchip,mclk-fs = <128>;
+		rockchip,cpu = <&i2s_2ch_0>;
+		rockchip,codec = <&dummy_codec>;
+		rockchip,format = "dsp_b";
+		rockchip,wait-card-locked = <0>;
+	};
+
 	spdif_tx_sound: spdif-tx-sound {
 		status = "disabled";
 		compatible = "simple-audio-card";
@@ -211,6 +222,10 @@
 	vccio5-supply = <&vcc_io>;
 };
 
+&i2s_2ch_0 {
+	status = "okay";
+};
+
 &i2s_8ch_2 {
 	status = "okay";
 };
-- 
2.35.3

